tests/estimator/test_compact_hamiltonian.py

```diff
@@ -0,0 +1,215 @@
+
+    assert VibrationalHamiltonian.__dataclass_params__.frozen is True
+    hamiltonian = VibrationalHamiltonian(num_modes=10, grid_size=30, taylor_degree=5)
+    with pytest.raises(AttributeError):
+        hamiltonian.num_modes = 20
+
+
+@pytest.mark.parametrize(
+    "num_modes, grid_size, taylor_degree, one_norm",
+    [(4, 10, 3, None), (16, 50, 2, 50.0), (8, 20, 4, 25)],
+)
+def test_vibrational_instantiation(num_modes, grid_size, taylor_degree, one_norm):
+    """Test successful instantiation with valid integer inputs."""
+    hamiltonian = VibrationalHamiltonian(
+        num_modes=num_modes, grid_size=grid_size, taylor_degree=taylor_degree, one_norm=one_norm
+    )
+
+    assert hamiltonian.num_modes == num_modes
+    assert hamiltonian.grid_size == grid_size
+    assert hamiltonian.taylor_degree == taylor_degree
+    assert hamiltonian.one_norm == one_norm
+
+
+@pytest.mark.parametrize(
+    "invalid_num_modes, invalid_grid_size, invalid_taylor_degree, invalid_one_norm",
+    [
+        ("4", 10, 3, None),
+        (16, 4.5, 2, None),
+        (16, 50, "2", None),
+        (4.5, 10, 3, None),
+    ],
+)
+def test_vibrational_invalid_types(
+    invalid_num_modes, invalid_grid_size, invalid_taylor_degree, invalid_one_norm
+):
+    """Test that TypeError is raised for invalid input types."""
+    with pytest.raises(TypeError):
+        VibrationalHamiltonian(
+            num_modes=invalid_num_modes,
+            grid_size=invalid_grid_size,
+            taylor_degree=invalid_taylor_degree,
+            one_norm=invalid_one_norm,
+        )
+
+
+def test_VibronicHamiltonian_is_frozen():
+    """Verify that the dataclass is immutable (frozen=True)."""
+
+    assert VibronicHamiltonian.__dataclass_params__.frozen is True
+    hamiltonian = VibronicHamiltonian(num_modes=10, num_states=2, grid_size=30, taylor_degree=5)
+    with pytest.raises(AttributeError):
+        hamiltonian.num_modes = 20
+
+
+@pytest.mark.parametrize(
+    "num_modes, num_states, grid_size, taylor_degree, one_norm",
+    [(4, 2, 10, 3, None), (16, 3, 50, 2, 50.0), (8, 5, 20, 4, 25)],
+)
+def test_vibronic_instantiation(num_modes, num_states, grid_size, taylor_degree, one_norm):
+    """Test successful instantiation with valid integer inputs."""
+    hamiltonian = VibronicHamiltonian(
+        num_modes=num_modes,
+        num_states=num_states,
+        grid_size=grid_size,
+        taylor_degree=taylor_degree,
+        one_norm=one_norm,
+    )
+
+    assert hamiltonian.num_modes == num_modes
+    assert hamiltonian.num_states == num_states
+    assert hamiltonian.grid_size == grid_size
+    assert hamiltonian.taylor_degree == taylor_degree
+    assert hamiltonian.one_norm == one_norm
+
+
+@pytest.mark.parametrize(
+    "invalid_num_modes, invalid_num_states, invalid_grid_size, invalid_taylor_degree, invalid_one_norm",
+    [
+        ("4", 2, 10, 3, None),
+        (16, 3.5, 50, 2, None),
+        (16, 3, "50", 2, None),
+        (4.5, 2, 10, 3, None),
```

Comment:
Not testing invalid Taylor degree or invalid one norm
```suggestion
        (4, 2, 10, 3.5, None),
        (4, 2, 10, 3, "None"),
```